storage/azure/src/main/java/io/aiven/kafka/tieredstorage/storage/azure/AzureBlobStorage.java

```diff
@@ -108,6 +114,9 @@ public long upload(final InputStream inputStream, final ObjectKey key) throws St
         parallelTransferOptions.setMaxSingleUploadSizeLong(blockSizeLong);
         final BlockBlobOutputStreamOptions options = new BlockBlobOutputStreamOptions()
             .setParallelTransferOptions(parallelTransferOptions);
+        // Be aware that metrics instrumentation is based on PutBlock (upload part)
+        // and PutBlockList (complete upload) used by this call.
+        // If upload changes, change metrics instrumentation accordingly.
```

Comment:
I just realized that if we upload a file smaller than `azure.upload.block.size`, it will be a plain `PutBlob`, i.e. no blocks involved. I think we should adapt to this as well.

Reply:
Good catch! adding a test for this, and including putBlob metric.

Reply:
The comment probably needs changing now :)